src/pyhf/contrib/utils.py

```diff
@@ -0,0 +1,67 @@
+"""Helper utilities for common tasks."""
+
+from urllib.parse import urlparse
+import tarfile
+from io import BytesIO
+import logging
+from .. import exceptions
+
+logging.basicConfig()
+log = logging.getLogger(__name__)
+
+try:
+    import requests
+
+    def download(archive_url, output_directory, force=False, compress=False):
+        """
+        Download the patchset archive from the remote URL and extract it in a
+        directory at the path given.
+
+        Example:
+
+            >>> from pyhf.contrib.utils import download
+            >>> download("https://www.hepdata.net/record/resource/1408476?view=true", "1Lbb-likelihoods")
+            >>> import os
+            >>> sorted(os.listdir("1Lbb-likelihoods"))
+            ['BkgOnly.json', 'README.md', 'patchset.json']
+            >>> download("https://www.hepdata.net/record/resource/1408476?view=true", "1Lbb-likelihoods.tar.gz", compress=True)
+            >>> import glob
+            >>> glob.glob("1Lbb-likelihoods.tar.gz")
+            ['1Lbb-likelihoods.tar.gz']
+
+        Args:
+            archive_url (`str`): The URL of the :class:`~pyhf.patchset.PatchSet` archive to download.
+            output_directory (`str`): Name of the directory to unpack the archive into.
+            force (`Bool`): Force download from non-approved host. Default is ``False``.
+            compress (`Bool`): Keep the archive in a compressed ``tar.gz`` form. Default is ``False``.
+
+        Raises:
+            :class:`~pyhf.exceptions.InvalidArchiveHost`: if the provided archive host name is not known to be valid
+        """
+        if not force:
+            valid_hosts = ["www.hepdata.net", "doi.org"]
```

Comment:
why do we care about valid hosts?

Reply:
I think in order to now have pyhf be a generic download tool .. i think constraining this for now is good, i.e. only allow "trustworthy" lhoods

Reply:
there is still --force

Reply:
What Lukas said.